### ISSUE TITLE
Is there any problem to change model float32 to float16

### ISSUE THREAD

**Issue by haonanzc** [mat10d/EvolvePro#14]
By changing the model float32 to float16 like below, I could generate embeddings on my local Mac with 32GB RAM and using esm2_t48_15B_UR50D, but if this change could harm the evolvepro result as the accuracy is decreased.

esm/extract.py
def run(args):
    model, alphabet = pretrained.load_model_and_alphabet(args.model_location)
    model = model.half()  #change float32 to float16, decrease the memory use
    model.eval()

**Comment by mat10d** [comment]
It may be slightly decreased, but this could be a smart workaround. Haven't tried it myself. You can also use a smaller esm model--you will see in our paper that there isn't a big drop off for the 650M model.
